src/sage/doctest/control.py

```diff
@@ -588,36 +588,38 @@ def __del__(self):
 
     def _init_warn_long(self):
         """
-        Pick a suitable default for the ``--warn-long`` option if not specified.
+        Pick a suitable default for the ``--warn-long`` option if not
+        specified.
 
         It is desirable to have all tests (even ``# long`` ones)
         finish in less than about 5 seconds. Longer tests typically
         don't add coverage, they just make testing slow.
 
-        The default used here is 60 seconds on a modern computer. It
-        should eventually be lowered to 5 seconds, but its best to
-        boil the frog slowly.
+        The default used here is 5 seconds, unless `--long` was used,
+        in which case it is 30 seconds.
 
-        The stored timings are used to adjust this limit according to
-        the machine running the tests.
+        TESTS:
 
-        EXAMPLES::
+        Ensure that the user's command-line options are not changed::
 
-            sage: from sage.doctest.control import DocTestDefaults, DocTestController
+            sage: from sage.doctest.control import (DocTestDefaults,
+            ....:                                   DocTestController)
             sage: DC = DocTestController(DocTestDefaults(), [])
             sage: DC.options.warn_long = 5.0
             sage: DC._init_warn_long()
-            sage: DC.options.warn_long    # existing command-line options are not changed
+            sage: DC.options.warn_long
             5.00000000000000
         """
         # default is -1.0
         if self.options.warn_long >= 0:     # Specified on the command line
             return
-        try:
-            self.options.warn_long = 60.0 * self.second_on_modern_computer()
-        except RuntimeError as err:
-            if not sage.doctest.DOCTEST_MODE:
-                print(err)   # No usable timing information
+
+        # The developer's guide says that even a "long time" test
+        # should ideally complete in under five seconds, so we're
+        # being rather generous here.
+        self.options.warn_long = 5.0
+        if self.options.long:
+            self.options.warn_long = 30.0
 
     def second_on_modern_computer(self):
```

Comment:
Yes, but it's a "public" function, so I didn't want to start an argument over the deprecation policy.

Reply:
I think it is safe to consider everything in doctest as private. But as a compromise, at least add a deprecation warning so that this can be removed in 1year?

Reply:
Done